apps/checker/app/matchers/DictionaryMatcher.scala

```diff
@@ -59,40 +59,45 @@ class DictionaryMatcher(
       entities: List[EntityInText]
   ): Boolean = {
     entities.exists(entity =>
-      entity.range.from == ruleMatch.fromPos && entity.range.to == ruleMatch.toPos
+      entity.range.from <= ruleMatch.fromPos && entity.range.to >= ruleMatch.toPos
     )
   }
 
+  def matchFallsWithinNamedEntity(
```

Comment:
Is there a reason we wouldn't also consider the position here when lining up found entities with text, as with `matchFallsWithinNamedEntityRange`? It removes the (presumably unlikely, but idk!) possibility of a word being used as a noun in one position, but another part of speech (e.g. verb) in another.

Reply:
Double checking that you've seen this – our approach has changed here, might make perfect sense but I want to make sure I understand why!

Reply:
Ahhh, I knew there was a reason we used ranges but had forgotten!

We changed it because we thought it might be weird if the same word gets marked as incorrect in one place and not another. It had the further benefit of simplifying the model (passing around `List[String]` instead of `List[EntityInText]`).

However, on reflection I think it makes more sense to use ranges, so I've reverted it here: https://github.com/guardian/typerighter/pull/453/commits/389754863022486ec35f087b4e236f369fba47bc

Reply:
Gotcha, yeah that's an interesting case – appreciate we're sorta at the mercy of our model here 😅